File: macos/Scanner/Sources/Scanner/Rules/AESMCP006.swift
public struct AESMCP006: Rule {
    public let id = "AES-MCP-006"
    public let severity: Severity = .medium

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        let uninstalledAppIds = Set(facts.appInstallations.filter { !$0.installed }.map(\.appId))
        let findings: [Finding] = facts.mcpServers.compactMap { server in
            guard server.disabled || uninstalledAppIds.contains(server.appId) else {
                return nil
            }
            return finding(app: server.appId, serverName: server.name, affectedPath: server.configPath)
        }
        return findings
    }
}

File: windows/Scanner/Rules/AESMCP006.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESMCP006 : IRule
{
    public string Id => "AES-MCP-006";
    public Severity Severity => Severity.Medium;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts)
    {
        var uninstalledAppIds = facts.AppInstallations
            .Where(installation => !installation.Installed)
            .Select(installation => installation.AppId)
            .ToHashSet(StringComparer.Ordinal);

        return facts.McpServers
            .Where(server => server.Disabled || uninstalledAppIds.Contains(server.AppId))
            .Select(server => this.Finding(server.AppId, server.Name, affectedPath: server.ConfigPath))
            .ToArray();
    }
}
